hal-core/src/mem/page.rs

```diff
@@ -249,7 +252,7 @@ impl<A: Address, S: Size> Page<A, S> {
 
     pub fn contains(&self, addr: impl Into<A>) -> bool {
         let addr = addr.into();
-        addr >= self.base && addr <= self.end_address()
+        addr >= self.base && addr <= self.end_addr()
     }
```

Comment:
if you make `end_addr` be non-inclusive again, this should be changed to be `addr < self.end_addr()`.